diff -Nrup a/Makefile b/Makefile
--- a/Makefile	2020-05-05 10:00:00.000000000 +0200
+++ b/Makefile	2020-05-26 19:11:01.173636175 +0200
@@ -61,11 +61,11 @@ all: $(TARGETS)
 	$(COMPILE.c) $(COMPILE_PIC) $< $(OUTPUT_OPTION)
 
 $(PROJECT): utempter.c diag.h
-	$(LINK.c) $(COMPILE_PIE) $(LINK_PIE) $(LINK_NOW) $(LDLIBS) $< $(OUTPUT_OPTION)
+	$(LINK.c) $(COMPILE_PIE) $(LINK_PIE) $(LINK_RELRO) $(LINK_NOW) $(LDLIBS) $< $(OUTPUT_OPTION)
 
 $(SHAREDLIB): iface.os $(MAP)
 	$(LINK.o) -shared \
-		-Wl,-soname,$(SONAME),--version-script=$(MAP),-z,defs \
+		-Wl,-soname,$(SONAME),--version-script=$(MAP),-z,relro,-z,defs \
 		-lc $< $(OUTPUT_OPTION)
 
 $(STATICLIB): iface.o
